libs/platform/src/lib/components/form/radio-group/radio/radio.component.ts

```diff
@@ -0,0 +1,120 @@
+import {
+    Component,
+    ChangeDetectorRef,
+    EventEmitter,
+    Input,
+    Output,
+    Optional,
+    Self,
+    ViewChild,
+    TemplateRef,
+    ChangeDetectionStrategy,
+} from '@angular/core';
+import { RadioButtonComponent as CoreRadioButtonComponent, stateType } from '@fundamental-ngx/core';
+import { NgControl, NgForm } from '@angular/forms';
+import { BaseInput } from '../../base.input';
+import { Status } from '../../form-control';
+
+let uniqueId = 0;
+
+@Component({
+    selector: 'fdp-radio-button',
+    templateUrl: './radio.component.html',
+    changeDetection: ChangeDetectionStrategy.OnPush,
+})
+export class RadioButtonComponent extends BaseInput {
+    /**
+     * Type of Radio buttons. valid | error | warning | default |information
+     * currently supported only error with BaseInput
+     */
+    @Input()
+    get state(): Status | stateType {
+        return this._state;
+    }
```

Comment:
Who or what will be setting a state in case of RB?

Reply:
changing the style for Error, which is _status in baseInput, It needs to be passed to Radio button. so i am doing [state]="status" in "radio-group.component.html".

Reply:
have removed state input and getting it from Status of BaseInput.